operations/admin_from_mdb.go

```diff
@@ -0,0 +1,282 @@
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"gopkg.in/mgo.v2/bson"
+)
+
+func toMdbForLocal() cli.Command {
+	const (
+		dbFlagName           = "db"
+		evergreenLocalDBName = "evergreen_local"
+		inputFlagName        = "input"
+		urlFlagName          = "url"
+	)
+	return cli.Command{
+		Name:  "restore",
+		Usage: "restore file produced by dump for a local Evergreen",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  urlFlagName,
+				Usage: "specify the mongodb url",
+				Value: "mongodb://127.0.0.1:27017",
+			},
+			cli.StringFlag{
+				Name:  dbFlagName,
+				Usage: fmt.Sprintf("write to this database"),
+				Value: evergreenLocalDBName,
+			},
+			cli.StringFlag{
+				Name:  inputFlagName,
+				Usage: "read data from this file",
+			},
+		},
+		Before: mergeBeforeFuncs(
+			func(c *cli.Context) error {
+				if c.String(inputFlagName) == "" {
+					return errors.New("must specify an input file")
+				}
+				return nil
+			},
+		),
+		Action: func(c *cli.Context) error {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			dbName := c.String(dbFlagName)
+			url := c.String(urlFlagName)
+			infn := c.String(inputFlagName)
+
+			client, err := mongo.NewClient(options.Client().ApplyURI(url))
+			if err != nil {
+				return errors.Wrap(err, "problem creating mongodb client")
+			}
+
+			connCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+			if err = client.Connect(connCtx); err != nil {
+				return errors.Wrap(err, "problem connecting to mongodb")
+			}
+
+			f, err := os.Open(infn)
+			if err != nil {
+				return errors.Wrapf(err, "problem opening file %s", infn)
+			}
+			defer f.Close()
+
+			gr, err := gzip.NewReader(f)
+			if err != nil {
+				return errors.Wrap(err, "problem creating gzip reader")
+			}
+
+			tr := tar.NewReader(gr)
+			for {
+				header, err := tr.Next()
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					return errors.Wrap(err, "problem iterating tar reader")
+				}
+
+				coll := client.Database(dbName).Collection(header.Name)
+				size, err := coll.CountDocuments(ctx, struct{}{})
+				if err != nil {
+					return errors.Wrap(err, "problem finding number of source documents")
+				}
+				if size > 0 && dbName != evergreenLocalDBName {
+					return errors.Errorf("looks like there are already documents in this collection ('%s'), and it's not in the %s database, exiting for safety", header.Name, evergreenLocalDBName)
+				}
+
+				var buf *bytes.Buffer
+				switch header.Typeflag {
+				case tar.TypeDir:
+					continue
+				case tar.TypeReg:
+					buf = &bytes.Buffer{}
+					io.Copy(buf, tr)
+				}
+
+				docs := []interface{}{}
+				for {
+					doc := &birch.Document{}
+					if _, err := doc.ReadFrom(buf); err != nil {
+						if err == io.EOF {
+							break
+						}
+						return errors.Wrap(err, "problem reading document from buffer")
+					}
+					docs = append(docs, doc)
+				}
+				coll.InsertMany(ctx, docs)
```

Comment:
this works as long as you're never inserting more documents than fit in ram, otherwise, you can do insertmany if larger than a given batch size...

Reply:
They already have to fit in RAM though, since we're putting them in a buffer in the io.Copy above.